### PR TITLE
Changed the namespace to exasol.nb_connector

### LINKED ISSUE #77: Move notebook connector into a separate namespace
Currently it uses `exasol` top level namespace, which might clutter with other packages.
Should be moved into separate unique package prefix, like `exasol.nb_connect` or somthing like this.

### PR DESCRIPTION
**Pull request by ahsimb**
closes #77

### CODE REVIEW

File: .github/workflows/checks.yml [hunk 1]
```diff
@@ -97,7 +97,20 @@ jobs:
...
+          large-packages: false
 
+      - name: Free disk space by removing large directories
+        run: |
+          sudo rm -rf /usr/local/graalvm/
+          sudo rm -rf /usr/local/.ghcup/
+          sudo rm -rf /usr/local/share/powershell
+          sudo rm -rf /usr/local/share/chromium
+          sudo rm -rf /usr/local/lib/node_modules
+          sudo rm -rf /opt/ghc
```
Comment (Nicoretti) [inline .github/workflows/checks.yml:113]: Interesting ... WTF is contained in this default images :exploding_head:

File: exasol/nb_connector/sagemaker_extension_wrapper.py [hunk 2]
```diff
@@ -1,20 +1,20 @@
 from exasol_sagemaker_extension.deployment.deploy_create_statements import DeployCreateStatements   # type: ignore
 from exasol_sagemaker_extension.deployment.sme_language_container_deployer import SmeLanguageContainerDeployer  # type: ignore
 
-from exasol.connections import (
+from exasol.nb_connector.connections import (
     get_external_host,
     open_pyexasol_connection,
 )
-from exasol.extension_wrapper_common import (
+from exasol.nb_connector.extension_wrapper_common import (
```
Comment (Nicoretti) [inline exasol/nb_connector/sagemaker_extension_wrapper.py:8]: The words `wrapper` and `common` do not provide much information.

**Review by Nicoretti — APPROVED** [review]
I like your approach to simplifying the module naming and structure for improved readability. Here are some suggestions that aim to further refine clarity and adherence to Python best practices:

Instead of:
```python
from exasol.nb_connector.ai_lab_config import AILabConfig as CKey

myconfig = CKey(...)
```
We could do:
```python
from exasol.nb_connector import ai_lab

myconfig = ai_lab.Config(...) 
```

## TL;DR
 generally we should use namespaces more to distinguish the functionality.

* `bfs_utils` -> `bucketfs`
* `itde_manager` -> `itde`
* ...

This approach eliminates the need for filler words such as 'manager', 'util', etc., thereby simplifying usage.

File: .github/workflows/checks.yml (hunk 1 above)
Comment (ahsimb) [reply .github/workflows/checks.yml:113]: No idea, I just copy/pasted it from the ai-lab repo. The test were no longer working without it.

**Comment by ahsimb** [comment]
Thanks for raising the good questions. I think it will be better to move them into a separate discussion possibly followed by separate tickets. This one is quite big as it is, at least as far as the reviewing is concerned. Plus there going to be subsequent changes in the notebooks in the AI-Lab.
